### PR TITLE
feat: add Hero icons and additional uptake of Seeds Icon

### PR DESCRIPTION
**Pull request by jaredcwhite**
## Issue Overview

This PR addresses #3397

- [x] This change addresses the issue in full
- [ ] This change addresses only certain aspects of the issue
- [ ] This change is a dependency for another issue
- [ ] This change has a dependency from another issue

## Description

This PR pull in the Seeds Icon component along with the revised standardization around using the Hero Icons library. In addition, it expands on work previously done by @cliu02 to set up an icon map in shared-helpers…we could take this a step forward perhaps with a React Context or something for full icon theming (tbd).

## How Can This Be Tested/Reviewed?

Various places where icons appear in Public and Partners sites should be reviewed. Of course the icons will be slightly different where there wasn't an exact match between the previous FontAwesome Free choice and the new Hero Icons choice.

## Checklist:

- [x] My code follows the style guidelines of this project
- [ ] I have added QA notes to the issue with applicable URLs
- [x] I have performed a self-review of my own code
- [x] I have reviewed the changes in a desktop view
- [ ] I have reviewed the changes in a mobile view
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules
- [x] I have assigned reviewers
- [ ] I have run `yarn generate:client` and/or created a migration if I made backend changes that require them
- [x] My commit message(s) is/are polished, and any breaking changes are indicated in the message and are well-described
- [ ] Commits made across packages purposefully have the same commit message/version change, else are separated into different commits

## Reviewer Notes:

Steps to review a PR:

- Read and understand the issue, and ensure the author has added QA notes
- Review the code itself from a style point of view
- Pull the changes down locally and test that the acceptance criteria is met
- Also review the acceptance criteria on the Netlify deploy preview (noting that these do not yet include any backend changes made in the PR)
- Either explicitly ask a clarifying question, request changes, or approve the PR if there are small remaining changes but the PR is otherwise good to go

## On Merge:

If you have one commit and message, squash. If you need each message to be applied, rebase and merge.

### CODE REVIEW

File: shared-helpers/src/views/CustomIconMap.tsx [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+import LockClosedIcon from "@heroicons/react/24/solid/LockClosedIcon"
+import ChevronLeftIcon from "@heroicons/react/20/solid/ChevronLeftIcon"
+import Clock from "@heroicons/react/24/outline/ClockIcon"
+import HomeModernIcon from "@heroicons/react/24/outline/HomeModernIcon"
+import EnvelopeIcon from "@heroicons/react/24/outline/EnvelopeIcon"
+
+export const CustomIconMap = {
+  application: Application,
+  profile: Profile,
+  lockClosed: <LockClosedIcon />,
```
Comment (emilyjablonski) [inline shared-helpers/src/views/CustomIconMap.tsx:12]: Curious why asome icons are mapped here and others pulled in in their respective files?

File: sites/public/src/components/applications/HouseholdMemberForm.tsx [hunk 2]
```diff
@@ -34,12 +35,9 @@ const HouseholdMemberForm = (props: HouseholdMemberFormProps) => {
             {props.strings?.edit ?? t("t.edit")}
           </Button>
         ) : (
-          <Icon
-            className="ml-2 pt-4 block"
-            size="medium"
-            symbol="lock"
-            fill={IconFillColors.primary}
-          />
+          <Icon className="ml-2 mt-4 text-primary" size="md">
```
Comment (emilyjablonski) [inline sites/public/src/components/applications/HouseholdMemberForm.tsx:38]: I think this is okay for now, but we're trying to not add any new tailwind styles as we move away from tailwind, and are adding module files as necessary. This won't be the seeds primary color. I know we chatted about it briefly but we might want some utility classes in seeds.

File: sites/public/src/pages/applications/preferences/general.tsx [hunk 3]
```diff
@@ -44,61 +43,24 @@ const ApplicationPreferencesGeneral = () => {
 
   return (
     <FormsLayout>
-      <FormCard header={<Heading priority={1}>{listing?.name}</Heading>}>
-        <ProgressNav
-          currentPageSection={currentPageSection}
-          completedSections={application.completedSections}
-          labels={conductor.config.sections.map((label) => t(`t.${label}`))}
-          mounted={OnClientSide()}
+      <Form onSubmit={handleSubmit(onSubmit)}>
```
Comment (emilyjablonski) [inline sites/public/src/pages/applications/preferences/general.tsx:46]: Ty for doing this!

**Review by emilyjablonski — APPROVED** [review]

File: shared-helpers/src/views/CustomIconMap.tsx (hunk 1 above)
Comment (jaredcwhite) [reply shared-helpers/src/views/CustomIconMap.tsx:12]: That's a good question…I probably was thinking about one-offs vs ones that get used in a number of different places. But we might want to map all of them, and also test the theory that they could easily be swapped out for new custom icons for certain jurisdictions (in a follow up PR I imagine)

**Pull request merged**